packages/mobile/src/web3/saga.ts

```diff
@@ -81,70 +71,36 @@ function* checkWeb3SyncProgressClaim() {
   }
 }
 
-// Checks both web3's claim for sync progress as well as checking the latest Block it returns
-function* checkSyncProgress() {
-  while (true) {
-    Logger.debug(TAG, 'Start checking web3 sync progress')
-
-    yield call(waitForGethConnectivity)
-    Logger.debug(TAG, 'Geth is connected')
-
-    const { web3SyncTimeout } = yield race({
-      web3SyncComplete: call(checkWeb3SyncProgressClaim),
-      web3SyncTimeout: delay(CHECK_WEB3_SYNC_PROGRESS_TIMEOUT),
-    })
-
-    if (web3SyncTimeout) {
-      Logger.error(TAG, 'checking web3 sync progress timed out')
-      continue
-    }
-
-    const latestBlock: Block = yield getLatestBlock()
-    if (latestBlock && latestBlock.number > 0) {
-      yield put(setLatestBlockNumber(latestBlock.number))
-      return
-    }
-
-    Logger.error(
-      TAG,
-      `web3 indicated sync complete, yet the latest block is ${JSON.stringify(latestBlock)}`
-    )
-  }
-}
-
 // The worker listening to sync progress requests
-function* checkSyncProgressWorker() {
-  while (true) {
+export function* checkWeb3Sync() {
```

Comment:
Could you add some tests for these functions?

Reply:
done!